### PR TITLE
Fix crashes for clients that don't support "workspace/configuration" requests

### PR DESCRIPTION
**Pull request by TwitchBronBron**
Prevent several crashes related to the language server incorrectly assuming that all clients support "workspace/configuration".

### CODE REVIEW

File: src/LanguageServer.spec.ts [hunk 1]
```diff
@@ -388,29 +389,6 @@ describe('LanguageServer', () => {
...
-
-            //no child bsconfig.json files, use the workspacePath
-            expect(
-                server.projects.map(x => x.projectPath)
-            ).to.eql([
-                workspacePath
-            ]);
-        });
-
-        it('ignores bsconfig.json files from vscode ignored paths', async () => {
```
Comment (TwitchBronBron) [inline src/LanguageServer.spec.ts:402]: Deleted this totally duplicate unit test

**Comment by ZeeD** [comment]
I know this is a bit unrelated to the PR, but do you think you can add another binary that expose the language server as a standalone application?
Or do you prefer to keep it outside the project?

**Comment by TwitchBronBron** [comment]
> I know this is a bit unrelated to the PR, but do you think you can add another binary that expose the language server as a standalone application? Or do you prefer to keep it outside the project?

Let's discuss this back in #665

**Review by chrisdp — APPROVED** [review]

File: src/LanguageServer.ts [hunk 2]
```diff
@@ -230,14 +230,18 @@ export class LanguageServer {
...
-        let config = await this.connection.workspace.getConfiguration({
-            scopeUri: workspaceFolder,
-            section: 'files'
-        }) as {
-            exclude: Record<string, boolean>;
+    private async getWorkspaceExcludeGlobs(workspaceFolder: string): Promise<string[]> {
+        let config = {
+            exclude: {} as Record<string, boolean>
         };
+        //if supported, ask vscode for the `files.exclude` configuration
```
Comment (KamasamaK) [inline src/LanguageServer.ts:237]: Shouldn't this comment use the generic "client" instead of specifically mentioning "vscode"?

Comment (TwitchBronBron) [reply src/LanguageServer.ts:237]: The `files` and `files.exclude` are specific to VSCode. I have no idea what other editors to do handle including/excluding files from the workspace, but they're _probably_ doing it differently than this. So, the comment still makes sense, and we can enhance the comment in the future as we add support for other editors.

Comment (KamasamaK) [reply src/LanguageServer.ts:237]: That's true that the specific configuration options are not part of the standard. Although there are other clients such as Theia that model their settings after VS Code.

**Pull request merged**